WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -376,16 +376,27 @@ private extension CollectOrderPaymentUseCase {
     ///
     func presentReceiptAlert(receiptParameters: CardPresentReceiptParameters, backButtonTitle: String, onCompleted: @escaping () -> ()) {
         // Present receipt alert
-        alerts.success(printReceipt: { [order, configuration] in
+        alerts.success(printReceipt: { [order, configuration, weak self] in
+            guard let self = self else { return }
+
             // Inform about flow completion.
             onCompleted()
 
             // Delegate print action
-            ReceiptActionCoordinator.printReceipt(for: order, params: receiptParameters, countryCode: configuration.countryCode)
+            ReceiptActionCoordinator.printReceipt(for: order,
+                                                  params: receiptParameters,
+                                                  countryCode: configuration.countryCode,
+                                                  cardReaderModel: self.connectedReader?.readerType.model,
+                                                  stores: self.stores,
+                                                  analytics: self.analytics)
+
+        }, emailReceipt: { [order, analytics, paymentOrchestrator, configuration, weak self] in
+            guard let self = self else { return }
 
-        }, emailReceipt: { [order, analytics, paymentOrchestrator, configuration] in
             // Record button tapped
-            analytics.track(event: .InPersonPayments.receiptEmailTapped(countryCode: configuration.countryCode))
+            analytics.track(event: .InPersonPayments
+                .receiptEmailTapped(countryCode: configuration.countryCode,
+                                    cardReaderModel: self.connectedReader?.readerType.model ?? ""))
```

Comment:
Good question! The reason I didn't make `cardReaderModel` optional like in `receiptPrint*` events is because emailing a receipt is only available after completing IPP successfully. After the IPP flow, it's still possible to print a receipt from "See receipt" in order details but there is no action to email a receipt. In `receiptEmail*` events, the `cardReaderModel` is supposed to be non-nil all the time but the `connectedReader` property is currently optional

Reply:
Lemme know if you think guarding on a non-nil `connectedReader` would make it more clear!